src/test/java/com/squareup/javapoet/TypeNameTest.java

```diff
@@ -69,10 +79,29 @@
     test(WildcardTypeName.supertypeOf(String.class), WildcardTypeName.supertypeOf(String.class));
   }
 
+  @Test public void equalsAndHashCodeAnnotated() {
+    test(TypeName.BYTE.annotated(list()), TypeName.BYTE.annotated(list()));
+    test(TypeName.INT.annotated(list(Override.class)), TypeName.INT.annotated(list(Override.class)));
+    test(ClassName.get(Map.class).annotated(list(AnnotationA.class, AnnotationB.class)),
+        ClassName.get(Map.class).annotated(list(AnnotationA.class, AnnotationB.class)));
+    assertNotEquals(TypeName.CHAR, TypeName.CHAR.annotated(list(Documented.class)));
+  }
+
   private void test(TypeName a, TypeName b) {
```

Comment:
Wanted the name to be in line with the other method names. Failed, though. Will try `@Test ... equalsAndHashCodeAnnotatedTypeName()`

Or are you referring to the name of the private `test` method?